build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -302,84 +287,34 @@ private void setDistributionType(ElasticsearchDistribution distribution, TestDis
         }
     }
 
-    // package protected so only TestClustersAware can access
-    @Internal
-    Collection<Configuration> getPluginAndModuleConfigurations() {
-        return pluginAndModuleConfigurations.values();
-    }
-
-    // creates a configuration to depend on the given plugin project, then wraps that configuration
-    // to grab the zip as a file provider
-    private Provider<RegularFile> maybeCreatePluginOrModuleDependency(String path, String consumingConfiguration) {
-        var configuration = pluginAndModuleConfigurations.computeIfAbsent(path, key -> {
-            var bundleDependency = this.project.getDependencies().project(Map.of("path", path, "configuration", consumingConfiguration));
-            return project.getConfigurations().detachedConfiguration(bundleDependency);
-        });
-
-        Provider<File> fileProvider = configuration.getElements()
-            .map(
-                s -> s.stream()
-                    .findFirst()
-                    .orElseThrow(
-                        () -> new IllegalStateException(consumingConfiguration + " configuration of project " + path + " had no files")
-                    )
-                    .getAsFile()
-            );
-        return project.getLayout().file(fileProvider);
-    }
-
     @Override
     public void plugin(Provider<RegularFile> plugin) {
         checkFrozen();
-        registerExtractedConfig(plugin);
         this.plugins.add(plugin.map(RegularFile::getAsFile));
     }
 
     @Override
     public void plugin(String pluginProjectPath) {
-        plugin(maybeCreatePluginOrModuleDependency(pluginProjectPath, "zip"));
+        throw new IllegalStateException("Not Supported API");
```

Comment:
Never used. We should not start now while migrating away from this Build logic

Reply:
I think this makes more sense to be an `UnsupportedOperationException`.

Reply:
fixed